src/js/components/Header/Header.js

```diff
@@ -1,19 +1,92 @@
-import React from 'react';
-
+import React, { useEffect, useState } from 'react';
 import { Box } from '../Box';
+import { useForwardedRef } from '../../utils';
+
+const Header = React.forwardRef(({ sticky, ...rest }, ref) => {
+  const containerRef = useForwardedRef(ref);
+  // show header as default then if scroll down showHeader is false
+  // else is scroll up showHeader is true.
+  const [height, setHeight] = useState();
+  const [width, setWidth] = useState();
+  const [right, setRight] = useState();
+  const [left, setLeft] = useState();
+  const [scrollUp, setScrollUp] = useState(false);
+
+  useEffect(() => {
+    //   // threshold inital value set to 0
+    const threshold = 0;
+    //   // number of pixels the document is currently scrolled
+    let lastScrollY = window.pageYOffset;
+    //   // run event listener callback once
+    let ticking = false;
+
+    // check if threshold is met which will
+    // determine if the user is scrolling up or down
+    // use ref here to get height and width
+
+    const updateScrollDir = () => {
+      const scrollY = window.pageYOffset;
+
+      if (Math.abs(scrollY - lastScrollY) < threshold) {
+        ticking = false;
+        return;
+      }
+      if (scrollY === 0) {
+        setScrollUp(false);
+        ticking = false;
+      } else setScrollUp(scrollY < lastScrollY);
+      // calcualte left, right, width, height
+      setHeight(`${containerRef.current.getBoundingClientRect().height}px`);
```

Comment:
Rather than storing these in separate state variables, how about putting them in a single object? For example:
`const [stickyStyles, setStickyStyles] = useState();`

Reply:
sounds good Ill clean this up the `width` was changing on me so I was going to look closer to that today

Reply:
I cleaned this section up